tests/test_app/test_data_store_new.py

```diff
@@ -30,8 +30,8 @@ def retain_nc_dstore(dstore):
     ]
     for i, item in enumerate(nc):
         dstore.write_not_completed(f"nc{i+1}", item.to_json())
-    assert 3 == len(dstore.not_completed)
-    assert 3 == len(list((ncdir / _MD5_TABLE).glob("*.txt")))
+    assert len(dstore.not_completed) == 3
```

Comment:
Delete `retain_nc_dstore()` and all calls to it.

Replace it by setting `pytest.fixture(scope="function")` ([see pytest docs](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html)). With that setting, pytest will tear down and build the fixture for each function call.